converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/RetryableFhirTransactionPipeline.cs

```diff
@@ -66,16 +68,20 @@ public Task ProcessAsync(ChangeFeedEntry changeFeedEntry, CancellationToken canc
             context[nameof(ChangeFeedEntry)] = changeFeedEntry;
 
             return _retryPolicy.ExecuteAsync(
-                (ctx, tkn) =>
+                async (ctx, tkn) =>
                 {
                     try
                     {
-                       return _fhirTransactionPipeline.ProcessAsync(changeFeedEntry, cancellationToken);
+                       await _fhirTransactionPipeline.ProcessAsync(changeFeedEntry, cancellationToken);
                     }
                     catch (TaskCanceledException ex) when (!cancellationToken.IsCancellationRequested)
                     {
                         throw new RetryableException(ex);
                     }
+                    catch (HttpRequestException ex)
```

Comment:
I think this makes sense to add to your policy instead, as the `TaskCanceledException` needed special-casing to differeniate cancellation scenarios

Reply:
Hmm so I added it here because in ChangeFeedProcessor I will otherwise need to catch HttpRequestExceptioin there as well. I thought it would be easier if I just wrap all exceptions that are retryable before so I can just catch retryable exception instead of each individual one.

Reply:
I see.

Yeah, you can rethrow it like that. I haven't thought through it completely it, but it may be more accurate to push this sort of logic directly into the retrying pipeline since some of this logic is based specifically on it. But that can be done another time

Reply:
Okay, Thanks!